common/src/channels.rs

```diff
@@ -819,10 +826,10 @@ async fn wrap_message_and_forward_to_network<
     network: &N,
     user_id_mapping: &HashMap<UserID, ecdsa::Public>,
     my_user_id: UserID,
-    associated_block_id: <WorkManager as WorkManagerInterface>::Clock,
-    associated_session_id: <WorkManager as WorkManagerInterface>::SessionID,
-    associated_retry_id: <WorkManager as WorkManagerInterface>::RetryID,
-    associated_task_id: <WorkManager as WorkManagerInterface>::TaskID,
+    associated_block_id: <TangleWorkManager as WorkManagerInterface>::Clock,
```

Comment:
Why is Tangle used here? Can this be generic over any blockchain?

Reply:
Yea, curious to sync on this. It does not seem correct semantically or in the right direction to have something `Tangle` named at this level. In fact there should be some config trait that captures most of these values and exposes functions to grab them like `retry`,`session`,`task`,`block_id`, or `timestamp` etc. that a task implementor can leverage and that don't all necessary need to be implemented for.

Reply:
This is a draft, not finished.

Reply:
We could, in its place (later down the road), encapsulate `retry,session,task,block_id, or timestamp` into some `EventMetadata` struct with a K,V hashmap (e.g., "retry -> int", "session" -> int ...). For now, my focus on doing one thing at a time, since decoupling our code is not an easy task